contracts/Registry.sol

```diff
@@ -238,72 +221,31 @@ contract Registry {
         }
     }
 
-    // ----------------
-    // TOKEN FUNCTIONS:
-    // ----------------
-
-    /**
-    @dev                Called by a voter to claim their reward for each completed vote. Someone
-                        must call updateStatus() before this can be called.
-    @param _challengeID The PLCR pollID of the challenge a reward is being claimed for
-    @param _salt        The salt of a voter's commit hash in the given poll
-    */
-    function claimReward(uint _challengeID, uint _salt) public {
-        // Ensures the voter has not already claimed tokens and challenge results have been processed
-        require(challenges[_challengeID].tokenClaims[msg.sender] == false);
-        require(challenges[_challengeID].resolved == true);
-
-        uint voterTokens = voting.getNumPassingTokens(msg.sender, _challengeID, _salt);
-        uint reward = voterReward(msg.sender, _challengeID, _salt);
-
-        // Subtracts the voter's information to preserve the participation ratios
-        // of other voters compared to the remaining pool of rewards
-        challenges[_challengeID].totalTokens -= voterTokens;
-        challenges[_challengeID].rewardPool -= reward;
-
-        // Ensures a voter cannot claim tokens again
-        challenges[_challengeID].tokenClaims[msg.sender] = true;
-
-        require(token.transfer(msg.sender, reward));
-
-        emit _RewardClaimed(_challengeID, reward, msg.sender);
-    }
-
-    /**
-    @dev                 Called by a voter to claim their rewards for each completed vote. Someone
-                         must call updateStatus() before this can be called.
-    @param _challengeIDs The PLCR pollIDs of the challenges rewards are being claimed for
-    @param _salts        The salts of a voter's commit hashes in the given polls
-    */
-    function claimRewards(uint[] _challengeIDs, uint[] _salts) public {
-        // make sure the array lengths are the same
-        require(_challengeIDs.length == _salts.length);
-
-        // loop through arrays, claiming each individual vote reward
-        for (uint i = 0; i < _challengeIDs.length; i++) {
-            claimReward(_challengeIDs[i], _salts[i]);
-        }
+    function resolveChallenge(bytes32 _listingHash) private {
+      Listing storage listing      = listings[_listingHash];
+      ChallengeInterface challenge = challengeForListingHash(_listingHash);
+      uint challengeID  = listings[_listingHash].challengeID;
+
+      // get the winner's reward
+      uint reward = challenge.tokenRewardAmount();
```

Comment:
I know it's using the same naming, but this seems to broad of a name. Wish it was something like `winnerRewardAmount()`